pkg/resource/deploy/deployment_executor.go

```diff
@@ -445,8 +449,36 @@ func (ex *deploymentExecutor) handleSingleEvent(event SourceEvent) error {
 	if err != nil {
 		return err
 	}
+	// Exclude the steps that depend on errored steps if ContinueOnError is set.
```

Comment:
Why is the deployment_executor responsible for this rather than the step generator, or even the source eval? We know the step generator/source eval will not see the new registration till the step_executor has finished it because it's dependent.